busybox/busybox.py

```diff
@@ -26,21 +26,29 @@ def harness(dut: Dut, ctx: TestContext, result: TestResult):
         parsed = dut.match.groupdict()
 
         if idx == 2:
-            msg.append("\t\t" + parsed["line"])
-            continue
+            # checks whether the message belongs to the next test
+            if "---Testcase" in parsed["line"]:
+                subresult = None
+                continue
```

Comment:
please create a separate regex pattern for the testcase initial log and add it to the `expect` method.